cmake/DLAF_AddTest.cmake

```diff
@@ -141,11 +141,25 @@ function(DLAF_addTest test_target_name)
 
     # APPLE platform does not support thread binding
     if (NOT APPLE)
-      list(APPEND _TEST_ARGUMENTS "--pika:use-process-mask")
+      # TODO pika:bind=none?
+      list(FILTER _PIKA_EXTRA_ARGS_LIST EXCLUDE REGEX "--pika:use-process-mask")
+      list(APPEND _PIKA_EXTRA_ARGS_LIST "--pika:use-process-mask")
     endif()
 
     if(NOT DLAF_TEST_THREAD_BINDING_ENABLED)
-      list(APPEND _TEST_ARGUMENTS "--pika:bind=none")
+      list(FILTER _PIKA_EXTRA_ARGS_LIST EXCLUDE REGEX "--pika:bind")
+      list(APPEND _PIKA_EXTRA_ARGS_LIST "--pika:bind=none")
+    endif()
+
+    if(IS_AN_MPI_TEST AND DLAF_MPI_PRESET STREQUAL "plain-mpi")
+      if(DLAF_CORE_PER_RANK GREATER_EQUAL 2)
```

Comment:
Is `DLAF_CORE_PER_RANK` guaranteed to be set? Or does the comparison gracefully handle it not being set...?

Reply:
I would expect it to be set whenever it is an MPI test or the user asked to run all test with MPI

https://github.com/eth-cscs/DLA-Future/blob/fadbecf2e427a896a1cb12a9d203ea12535c5eb9/cmake/DLAF_AddTest.cmake#L114-L118

Reply:
Ok. I'm not sure I understand all the logic behind the MPI options, but what about when `MPIEXEC_NUMCORE_FLAG` isn't set? I guess in that case you can't set the number of cores per rank anyway... but is this check: https://github.com/eth-cscs/DLA-Future/blob/fadbecf2e427a896a1cb12a9d203ea12535c5eb9/cmake/DLAF_AddTest.cmake#L107 required here as well?

Reply:
Yeah, actually you're right. `DLAF_CORE_PER_RANK` has the fallback to 1 just in that branch when `MPIEXEC_NUMCORE_FLAG` exists.

The logic about option `--pika:threads` applies just for `plain-mpi` preset, which doesn't have neither `MPIEXEC_NUMCORE_FLAG` nor `MPIEXEC_NUMCORES`. This means that we would always end up not having `DLAF_CORE_PER_RANK` set, but moreover we end up not having information about how many cores there are on the system.

Should we just go for setting it to 2 without any clever logic inspecting number of cores available?

@rasolca @msimberg @teonnik

Reply:
What about `MPIEXEC_MAX_NUMPROCS`? Apparently it defaults to the number of cores.